### PR TITLE
Run3-TB46 Fix the test beam runs in view of present changes

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Fix 2006 test beam configuration in view of recent changes in accessing the geometry parameters

#### PR validation:

Tested with many cfg files in the area SimG4CMS/HcalTestBeam/test/python. The changes are supposed not to affect standard workflows

#### if this PR is a backport please specify the original PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
@bsunanda , thanks, it is running fine to me.

File: SimG4CMS/Calo/src/HCalSD.cc [hunk 1]
```diff
@@ -170,47 +177,46 @@ HCalSD::HCalSD(const std::string& name,
...
+      for (auto lvol : *lvs) {
+        if (lvol->GetName() == namv) {
+          lv = lvol;
+          break;
+        }
       }
-    }
-    hfLV.push_back(lv);
-    hfLevels.push_back(temp[i]);
+      hfLV.push_back(lv);
```
Comment (ianna) [inline SimG4CMS/Calo/src/HCalSD.cc]:
```
emplace_back
```

File: SimG4CMS/Calo/src/HCalSD.cc [hunk 2]
```diff
@@ -170,47 +177,46 @@ HCalSD::HCalSD(const std::string& name,
...
+      hfLevels.push_back(temp[i]);
```
Comment (ianna) [inline SimG4CMS/Calo/src/HCalSD.cc]:
```
emplace_back
```

**Review by ianna — CHANGES_REQUESTED** [review]

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by ianna** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by fabiocos** [comment]
+1

@bsunanda I understand that the test beam code behaviour has been tested in standalone studies

**Pull request merged**